src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,57 @@
+import { SignUpDto } from './dto/auth-signup.dto';
+import { LoginDto } from './dto/auth-login.dto';
+import { User } from '../user/user.entity';
+import { jwtService } from '../../shared/utils/JwtService';
+import { Repository } from 'typeorm';
+import uuid from "uuid";
+import bcrypt from 'bcrypt';
+
+export class AuthService {
+  constructor(private userRepository: Repository<User>){}
+
+  async signup(signUpDto: SignUpDto) {
+    const { username, email, password } = signUpDto;
+
+    const existingUser = await this.userRepository.findOne({ where: { email } });
+    if (existingUser) {
+      return `This user already exists!`
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+    const newUserId = uuid.v4()
+
+    const newUser = this.userRepository.create({
+      id: newUserId,
+      username,
+      email,
+      password: hashedPassword,
+    });
+
+    const tokens = jwtService.generateTokens({ id: newUser.id, email: newUser.email });
+    newUser.refreshToken = tokens.refreshToken;
+
+    await this.userRepository.save(newUser);
+
+    return {
+      ...newUser,
```

Comment:
be more careful when returning user data. sensitive data should not be sent to user like password